easybuild/tools/toolchain/toolchain.py

```diff
@@ -269,6 +270,10 @@ def as_dict(self, name=None, version=None):
             name = self.name
         if version is None:
             version = self.version
+        if self.hide_deps:
```

Comment:
no need for this check if you make sure `self.hide_deps` is always a list (by using `[]` as a fallback, see above)